backend/src/test/java/com/engineering/thesis/backend/controllerIntegrationTests/AppointmentEPTests.java

```diff
@@ -0,0 +1,191 @@
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@ExtendWith(MockitoExtension.class)
+@WebMvcTest(value = AppointmentController.class)
+class AppointmentEPTests extends MockConfiguration {
+
+    @MockBean
+    private AppointmentServiceImpl appointmentService;
+
+    @Test
+    public void selectAllShouldReturnAppointmentWhenProperRoleIsSelected() throws Exception {
+        final User userDoctor1 = new User(1l, "Tom", "Kowalsky", "dsadzx1xczsa@osom.com", "1I@wsdas", "ROLE_DOCTOR", true);
+        final User userDoctor2 = new User(2l, "Tom", "Kowalsky", "dsadzx2xczsa@osom.com", "1I@wsdas", "ROLE_DOCTOR", true);
+        final Doctor doctor1 = new Doctor(1l, userDoctor1, "Cardiology");
+        final Doctor doctor2 = new Doctor(2l, userDoctor2, "Cardiology");
+        final User userPatient1 = new User(4l, "Tom", "Kowalsky", "dsad12ssssa@osom.com", "1I@wsdas", "ROLE_PATIENT", true);
+        final User userPatient2 = new User(5l, "Tom", "Kowalsky", "dsad32ssssa@osom.com", "1I@wsdas", "ROLE_PATIENT", true);
+        final Patient patient1 = new Patient(1l, userPatient1, true);
+        final Patient patient2 = new Patient(2l, userPatient2, true);
+        when(appointmentService.selectAll())
+                .thenReturn(List.of(
+                        new Appointment(1l, patient1, doctor1, 1000L, LocalDateTime.now()),
+                        new Appointment(2l, patient2, doctor2, 2000, LocalDateTime.now())
+                ));
+        RulesMap().forEach((name, Role) -> {
+            try {
+                System.out.println("Performing select appointment with role: " + name);
+                if (name == "Doctor" || name == "Patient") {
+                    this.mockMvc
+                            .perform(MockMvcRequestBuilders.get("/api/appointment/select").with(Role))
+                            .andExpect(status().isOk())
+                            .andExpect(MockMvcResultMatchers.jsonPath("$.size()").value(2))
+                            .andExpect(MockMvcResultMatchers.jsonPath("$[1].cost").value(2000))
+                            .andExpect(MockMvcResultMatchers.jsonPath("$[0].cost").value(1000));
+                }
+                if (name == "Invalid") {
+                    this.mockMvc
+                            .perform(MockMvcRequestBuilders.get("/api/appointment/select").with(Role))
+                            .andExpect(status().isForbidden());
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        });
+    }
+
+    @Test
+    public void selectByIdShouldReturnAppointmentWhenProperRoleIsSelected() throws Exception {
+        final User userDoctor1 = new User(1l, "Tom", "Kowalsky", "dsadzx1xczsa@osom.com", "1I@wsdas", "ROLE_DOCTOR", true);
```

Comment:
to samo  w tym miejscu polecam użycie builder'a jak i w kolejnych miejscach gdzie tworzysz nowe obiekty do testów